


# Testing for tissue heterogeneity
## Tissue signatures {#sec:signatures}
In section \@ref(validating-signatures), we identified a set of
9 *reference signatures* (table \@ref(tab:refsig)) which unambiguously
identify their corresponding tissue across platforms and species.


Table: (\#tab:refsig)reference signatures

Reference Signature    Tissue          
---------------------  ----------------
GTEX_Blood             blood           
GTEX_Brain             brain           
GTEX_Heart             heart           
GTEX_Kidney            kidney          
GTEX_Liver             liver           
GTEX_Pancreas          pancreas        
GTEX_Muscle_Skeletal   skeletal muscle 
GTEX_Skin              skin            
GTEX_Testis            testis          

## Data {#sec:data_selection}

In section \@ref(sample-selection) we show in detail, how we obtain
and filter data from the Gene Expression Omnibus.
Table \@ref(tab:filteringprocess) shows a summary of this process, listing
the remaining samples and studies after each filtering step.


Table: (\#tab:filteringprocess)Selecting samples from GEO

filtering step                            samples   studies
---------------------------------------  --------  --------
total (as of 2016-12-07)                  1945417     73719
tissue annotated                           760798     24267
gene symbols annotated                     275206      9632
human orthologous gene symbols             253210      8083
Single-channel microarray                  235237      7561
tissue mapped to controlled vocabulary     136230      3800
human, rat, mouse                          134334      3757
minimal inter-sample variance               96074      3129
housekeeping signatures                     81845      2926
reference signature available               51948      1760

## Testing samples for heterogeneity
We tested for enrichment of
97
selected signatures provided by BioQC, the
9
reference signatures generated by us and one random control signature of
100 randomly drawn genes on all 51948 selected samples
resulting in a list of 5610384 (sample, signature, pvalue) pairs.
This process is described in detail in section \@ref(sample-processing).

Our intention is to identify samples that show *tissue heterogeneity*,
*i.e.* unintentional profiling of cells of other origin than the target tissue
of profiling. We classify samples as *not heterogenous* or *heterogenous*.
We call a classification *true-positive* if the given sample is classified
as *heterogenous* and the sample indeed contains cells different from the
annotated tissues. Analogous, we call a classification *false-positive*
if the given sample is classified as *heterogenous* but in reality only contains cells from the annotated tissue.

Naively, we would label a sample as heterogenous, if a signature unrelated
to the annotated tissue exceeds a certain score. The problem with this
approach is, that some signatures overlap; the resulting scores are therefore
correlated and will lead to false-positives. One cannot simply solve this
problem by excuding genes that are members of multiple signatures, as it is
easily possible to build two (in fact many) distinct, non-overlapping
signatures matching the same tissue.

In section \@ref(validating-signatures) we haven created toroughly validated
*reference signatures* for 9 tissues. Even though we have demonstrated that
each signature unambigously identifies its corresponding tissue (*i.e.* scores
highest), the signatures could still be correlated. Some of them in fact are,
e.g. cardiac muscle and skeletal muscle (see figure
\@ref(fig:correlationexample)). Moreover, we lack sufficient data to perform an
independent-sample validation on the signatures provided by BioQC. Accounting
for correlation with the reference signatures, we can still avoid
false-positives, even if a signature is characteristic for a tissue
histologically close to the annotated tissue.

We therefore take the following approach to account for the correlation of signatures:
A given sample $s$ annotated as tissue $t$ is tested for enrichment with signature $k_{\text{test}}$ resulting in a p-value $p_{\text{test}}$. Let $k_{\text{ref}}$ be the reference signature associated with tissue $t$ and $p_{\text{ref}}$ the p-value of testing $s$ for enrichment of $k_{\text{ref}}$.

(1) If the Benjamini-Hochberg-adjusted $p_{\text{test}} >= 0.01$, we
assume that $s$ is not heterogenous; else continue.
(2) We fit a linear model using `rlm` from the `R` `MASS` package of
$|log10(p_{\text{test}})|$ against $|log10(p_{\text{ref}})|$ for all samples
annotated as $t$. We assume that the residuals $R$ of the linear model follow
a normal distribution $R \sim \mathcal{N}(0, \sigma^2)$, where $\sigma$ is the
standard deviation of the residuals.
(3) We extract the residual $r$ corresponding to sample $s$. We calculate the
p-value $p_{\text{corr}} = 1 - \text{CDF}_R(r)$ where $\text{CDF}_R$ is the
cumulative density function of $\mathcal{N}(0, \sigma^2)$.
(4) If the Bonferroni-adjusted $p_{\text{corr}} < 0.01$, we reject the
hypothesis that $k_{\text{test}}$ is enriched only due to correlation and label
the sample as heterogenous.

<div class="figure">
<img src="40_heterogeneity_test_files/figure-html/correlationexample-1.png" alt="Examples of signature correlation. Panels A-C: correlation of the signature scores (y-axis) against the scores of a reference signature (x-axis). The purple line indicates the model fitted to the data, the shaded area the 3*sigma confidence interval. Points are colored according to whether they would be discarded already in the FDR-filtering step. (A) Brain scores of kidney samples against the scores of the kidney signature. There is virtually no correlation, nor are there outliers. Appraently, no kidney samples are contaminated with neuronal cells. (B) Liver scores of kidney samples against scores of the kidney signature. The samples are not correlated, however some outliers are detected which are samples potentially containing liver cells. (C) Cardiac muscle scores of skeletal muscle samples against skeletal muscle scores. The scores are highly correlated. While most of the poinst exceed the FDR threshold, they will not be classified as outliers because the elevated score can be explained with the correlation. Panels D and E show the boxplots of the scores of various signatures on kidney and heart samples respectively. " width="1152" />
<p class="caption">(\#fig:correlationexample)Examples of signature correlation. Panels A-C: correlation of the signature scores (y-axis) against the scores of a reference signature (x-axis). The purple line indicates the model fitted to the data, the shaded area the 3*sigma confidence interval. Points are colored according to whether they would be discarded already in the FDR-filtering step. (A) Brain scores of kidney samples against the scores of the kidney signature. There is virtually no correlation, nor are there outliers. Appraently, no kidney samples are contaminated with neuronal cells. (B) Liver scores of kidney samples against scores of the kidney signature. The samples are not correlated, however some outliers are detected which are samples potentially containing liver cells. (C) Cardiac muscle scores of skeletal muscle samples against skeletal muscle scores. The scores are highly correlated. While most of the poinst exceed the FDR threshold, they will not be classified as outliers because the elevated score can be explained with the correlation. Panels D and E show the boxplots of the scores of various signatures on kidney and heart samples respectively. </p>
</div>
