<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)</title>
  <meta name="description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity”." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity”." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)" />
  
  <meta name="twitter:description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity”." />
  

<meta name="author" content="Gregor Sturm, Markus List, Jitao David Zhang" />


<meta name="date" content="2020-11-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="validating-signatures.html"/>
<link rel="next" href="testing-for-tissue-heterogeneity.html"/>
<script src="libs/header-attrs-2.4/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this document</a></li>
<li class="chapter" data-level="2" data-path="validating-signatures.html"><a href="validating-signatures.html"><i class="fa fa-check"></i><b>2</b> Validating Tissue Signatures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="validating-signatures.html"><a href="validating-signatures.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-validation-of-signatures-on-the-gtex-dataset"><i class="fa fa-check"></i><b>2.2</b> Cross-Validation of signatures on the GTEx dataset</a></li>
<li class="chapter" data-level="2.3" data-path="validating-signatures.html"><a href="validating-signatures.html#reference-signatures"><i class="fa fa-check"></i><b>2.3</b> ‘Reference’ Signatures</a></li>
<li class="chapter" data-level="2.4" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-platform-cross-species-validation"><i class="fa fa-check"></i><b>2.4</b> Cross-Platform Cross-Species Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="curating-data.html"><a href="curating-data.html"><i class="fa fa-check"></i><b>3</b> Sample data and metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="curating-data.html"><a href="curating-data.html#geo"><i class="fa fa-check"></i><b>3.1</b> GEO</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="curating-data.html"><a href="curating-data.html#downloading-geo-data"><i class="fa fa-check"></i><b>3.1.1</b> Downloading GEO data</a></li>
<li class="chapter" data-level="3.1.2" data-path="curating-data.html"><a href="curating-data.html#filtering-geo-data"><i class="fa fa-check"></i><b>3.1.2</b> Filtering GEO data</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="curating-data.html"><a href="curating-data.html#archs4"><i class="fa fa-check"></i><b>3.2</b> ARCHS4</a></li>
<li class="chapter" data-level="3.3" data-path="curating-data.html"><a href="curating-data.html#normalize-tissues"><i class="fa fa-check"></i><b>3.3</b> Normalize Tissue Names</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html"><i class="fa fa-check"></i><b>4</b> Testing for tissue heterogeneity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:signatures"><i class="fa fa-check"></i><b>4.1</b> Tissue signatures</a></li>
<li class="chapter" data-level="4.2" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#testing-samples-for-heterogeneity"><i class="fa fa-check"></i><b>4.2</b> Testing samples for heterogeneity</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#determining-tau."><i class="fa fa-check"></i><b>4.2.1</b> Determining <span class="math inline">\(\tau\)</span>.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>5</b> Results</a>
<ul>
<li class="chapter" data-level="5.1" data-path="results.html"><a href="results.html#heterogeneity-using-gtex-signatures-only"><i class="fa fa-check"></i><b>5.1</b> Heterogeneity using GTEx signatures only</a></li>
<li class="chapter" data-level="5.2" data-path="results.html"><a href="results.html#heterogeneity-using-all-signatures"><i class="fa fa-check"></i><b>5.2</b> Heterogeneity using all signatures</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figures-for-publication.html"><a href="figures-for-publication.html"><i class="fa fa-check"></i><b>6</b> Figures for publication</a></li>
<li class="chapter" data-level="7" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>7</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="curating-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Sample data and metadata</h1>
<p>In this section, we describe how we obtained and curated gene expression data and sample metadata.</p>
<div id="geo" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> GEO</h2>
<div id="downloading-geo-data" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Downloading GEO data</h3>
<p>We retrieved sample metadata for GEO using the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/GEOmetadb/inst/doc/GEOmetadb.html">GEOmetadb</a> package.
We download the studies with <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOquery</a> and store them as R <a href="https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf">ExpressionSet</a> using the R script <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R">geo_to_eset.R</a>. We used the <code>annotGPL=TRUE</code> option of <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOquery</a>’s <code>getGEO</code> function to obtain gene symbols for the studies, where available.
Since the tissue signatures use human gene symbols, we added human orthologs for all mouse and rat samples using the <a href="https://github.com/Accio/ribios">ribiosAnnotation</a> package.</p>
</div>
<div id="filtering-geo-data" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Filtering GEO data</h3>
<p>We filtered GEO samples by the following criteria:</p>
<ol style="list-style-type: decimal">
<li>the tissue or origin is annotated,</li>
<li>gene symbols are annotated,</li>
<li>the readout was performed on a single-channel microarray, and</li>
<li>the tissue could be mapped to our <a href="curating-data.html#normalize-tissues">controlled vocabulary</a> (CV).</li>
<li>We only retained samples from the three major organisms: human, rat and mouse.</li>
<li>We removed studies which have been normalized per-gene and where ubiquitous house-keeping genes were not expressed.</li>
<li>Finally, we only retained samples originating from tissues for which a <em>reference signature</em> is available.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:funnelgeo"></span>
<img src="../figures/funnel_geo.png" alt="Summary of filtering steps on GEO samples" width="40%" />
<p class="caption">
Figure 3.1: Summary of filtering steps on GEO samples
</p>
</div>
</div>
</div>
<div id="archs4" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> ARCHS4</h2>
<p>In addition to GEO, we used data from
<a href="https://amp.pharm.mssm.edu/archs4/">ARCHS4</a>, a publicly available data
collection of annotaed, consistently processed gene expression datasets based
on RNA-sequencing. We downloaded gene expression and metadata as <code>RData</code> objects
from the <a href="https://amp.pharm.mssm.edu/archs4/download.html">ARCHS4 website</a> (version 8.0).</p>
<p>We filtered samples by the following criteria:</p>
<ol style="list-style-type: decimal">
<li>The library is a transcriptomic cDNA library, the library strategy is RNA-seq, and either polyA or total RNA were extracted.</li>
<li>We excluded single-cell RNA-seq samples (none of the annotation fields may contain the keywords “single-cell”, “single cell” or “smartseq”)</li>
<li>At least 500,000 reads could be mapped to genes.</li>
<li>The tissue could be mapped to our <a href="curating-data.html#normalize-tissues">controlled vocabulary</a> (CV).</li>
<li>Finally, we only retained samples originating from tissues for which a <em>reference signature</em> is available.</li>
</ol>
<p>Gene counts were normalized into TPM values before analysing them with BioQC.</p>
<div class="figure" style="text-align: center"><span id="fig:funnelarchs"></span>
<img src="../figures/funnel_archs4.png" alt="Summary of filtering steps on ARCHS4 samples" width="40%" />
<p class="caption">
Figure 3.2: Summary of filtering steps on ARCHS4 samples
</p>
</div>
</div>
<div id="normalize-tissues" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Normalize Tissue Names</h2>
<p>The annotation of tissues is inconsistent within GEO. A “liver” sample can be termed <em>e.g.</em> “liver”, “liver biopsy” or “primary liver”. We, therefore, need a way to <em>normalize</em> the tissue name. We manually mapped the most abundant tissues to a controlled vocabulary in this <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx">Excel sheet</a>.</p>
<p>In order to find out which samples show <em>tissue heterogeniety</em>, we first need to define which signatures we would expect in a certain tissue. We mapped signatures to the respective tissue type in this
<a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx">Excel sheet</a>. For example, we map the signatures <code>Intestine_Colon_cecum_NR_0.7_3</code> and <code>Intestine_Colon_NR_0.7_3</code> to <em>colon</em>.</p>
<p>Since the “reference signatures” are not as specific as the annotation in the GEO, we created <em>tissue sets</em> to combine them into groups. For instance, it is hard to distinguish <em>jejunum</em> from <em>colon</em>, but easy to distinguish the two from other tissues. We therefore created a tissue set <em>intestine</em>, which contains both <em>jejunum</em> and <em>colon</em> and references all signatures associated with the two tissues. This information is part of the same <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx">Excel sheet</a>.</p>
<div style="page-break-after: always;"></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="validating-signatures.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="testing-for-tissue-heterogeneity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
