<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)</title>
  <meta name="description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity.”" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity.”" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Sample data and metadata | Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)" />
  
  <meta name="twitter:description" content="In this study, we systematically search GEO and ARCHS4 for cases of “tissue heterogeneity.”" />
  

<meta name="author" content="Gregor Sturm, Markus List, Jitao David Zhang" />


<meta name="date" content="2021-06-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="validating-signatures.html"/>
<link rel="next" href="testing-for-tissue-heterogeneity.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About this document</a></li>
<li class="chapter" data-level="2" data-path="validating-signatures.html"><a href="validating-signatures.html"><i class="fa fa-check"></i><b>2</b> Validating Tissue Signatures</a>
<ul>
<li class="chapter" data-level="2.1" data-path="validating-signatures.html"><a href="validating-signatures.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-validation-of-signatures-on-the-gtex-dataset"><i class="fa fa-check"></i><b>2.2</b> Cross-Validation of signatures on the GTEx dataset</a></li>
<li class="chapter" data-level="2.3" data-path="validating-signatures.html"><a href="validating-signatures.html#reference-signatures"><i class="fa fa-check"></i><b>2.3</b> Reference Signatures</a></li>
<li class="chapter" data-level="2.4" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-platform-cross-species-validation"><i class="fa fa-check"></i><b>2.4</b> Cross-Platform Cross-Species Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="curating-data.html"><a href="curating-data.html"><i class="fa fa-check"></i><b>3</b> Sample data and metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="curating-data.html"><a href="curating-data.html#normalize-tissues"><i class="fa fa-check"></i><b>3.1</b> Standardize Tissue Names</a></li>
<li class="chapter" data-level="3.2" data-path="curating-data.html"><a href="curating-data.html#geo"><i class="fa fa-check"></i><b>3.2</b> GEO</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="curating-data.html"><a href="curating-data.html#downloading-geo-data"><i class="fa fa-check"></i><b>3.2.1</b> Downloading GEO data</a></li>
<li class="chapter" data-level="3.2.2" data-path="curating-data.html"><a href="curating-data.html#filtering-geo-data"><i class="fa fa-check"></i><b>3.2.2</b> Filtering GEO data</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="curating-data.html"><a href="curating-data.html#archs4"><i class="fa fa-check"></i><b>3.3</b> ARCHS4</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html"><i class="fa fa-check"></i><b>4</b> Testing for tissue heterogeneity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:signatures"><i class="fa fa-check"></i><b>4.1</b> Tissue signatures</a></li>
<li class="chapter" data-level="4.2" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#testing-samples-for-heterogeneity"><i class="fa fa-check"></i><b>4.2</b> Testing samples for heterogeneity</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="additional-results.html"><a href="additional-results.html"><i class="fa fa-check"></i><b>5</b> Additional Results</a>
<ul>
<li class="chapter" data-level="5.1" data-path="additional-results.html"><a href="additional-results.html#distribution-of-heterogeneity-across-studies"><i class="fa fa-check"></i><b>5.1</b> Distribution of heterogeneity across studies</a></li>
<li class="chapter" data-level="5.2" data-path="additional-results.html"><a href="additional-results.html#heterogeneity-along-time"><i class="fa fa-check"></i><b>5.2</b> Heterogeneity along time</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tissue heterogeneity is prevalent in gene expression studies (Supplementary Information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="curating-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Sample data and metadata</h1>
<p>In this section, we describe how we obtained and curated gene expression data and sample metadata.</p>
<div id="normalize-tissues" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Standardize Tissue Names</h2>
<p>The annotation of tissues is inconsistent within GEO and ARCHS4. A “liver” sample can be termed <em>e.g.</em> “liver,” “liver biopsy” or “primary liver.” We, therefore, need a way to <em>standardize</em> the tissue name. We manually mapped the most abundant tissues to a controlled vocabulary. Next, in order to find out which samples are subject to <em>tissue heterogeniety</em>, we first need to define which signatures we would expect in a certain tissue. For example, we map the signatures <code>Intestine_Colon_cecum_NR_0.7_3</code> and <code>Intestine_Colon_NR_0.7_3</code> to <em>colon</em>.
Since the reference signatures are not as specific as the tissue annotation, we created <em>tissue sets</em> to combine them into groups. For instance, it is hard to distinguish <em>jejunum</em> from <em>colon</em>, but easy to distinguish the two from other tissues. We therefore created a tissue set <em>intestine</em>, which contains both <em>jejunum</em> and <em>colon</em> and references all signatures associated with the two tissues. All of the described mappings are available from <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx">Supplementary Table 1</a>.<a href="#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a></p>
</div>
<div id="geo" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> GEO</h2>
<div id="downloading-geo-data" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Downloading GEO data</h3>
<p>We retrieved sample metadata for GEO using the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/GEOmetadb/inst/doc/GEOmetadb.html">GEOmetadb</a> package <span class="citation">(<a href="#ref-geometadb" role="doc-biblioref">Zhu et al. 2008</a>)</span>.
We downloaded the studies with <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOquery</a> <span class="citation">(<a href="#ref-geoquery" role="doc-biblioref">Davis and Meltzer 2007</a>)</span> and stored them as R <a href="https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf">ExpressionSet</a> <span class="citation">(<a href="#ref-Huber2015" role="doc-biblioref">Huber et al. 2015</a>)</span> using the R script <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R">geo_to_eset.R</a>.<a href="#fn10" class="footnote-ref" id="fnref10"><sup>10</sup></a> We used the <code>annotGPL=TRUE</code> option of GEOquery’s <code>getGEO</code> function to obtain gene symbols for the studies, where available.
Since the tissue signatures use human gene symbols, we added human orthologs for all mouse and rat samples.</p>
</div>
<div id="filtering-geo-data" class="section level3" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Filtering GEO data</h3>
<p>We filtered GEO samples by the following criteria (figure <a href="curating-data.html#fig:funnelgeo">3.1</a>):</p>
<ol style="list-style-type: decimal">
<li>the tissue or origin is annotated,</li>
<li>gene symbols are annotated,</li>
<li>the readout was performed on a single-channel microarray, and</li>
<li>the tissue could be mapped to the controlled vocabulary (section <a href="curating-data.html#normalize-tissues">3.1</a>).</li>
<li>We only retained samples from the three major organisms: human, rat, mouse.</li>
<li>We removed studies which have been normalized per-gene and where ubiquitous house-keeping genes were not expressed.</li>
<li>Finally, we only retained samples originating from tissues for which a <em>reference signature</em> is available.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:funnelgeo"></span>
<img src="/home/sturm/projects/2019/bioqc_geo/figures/funnel_geo.png" alt="Summary of filtering steps on GEO samples" width="40%" />
<p class="caption">
Figure 3.1: Summary of filtering steps on GEO samples
</p>
</div>
</div>
</div>
<div id="archs4" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> ARCHS4</h2>
<p>In addition to GEO, we used data from
<a href="https://amp.pharm.mssm.edu/archs4/">ARCHS4</a> <span class="citation">(<a href="#ref-Lachmann2018" role="doc-biblioref">Lachmann et al. 2018</a>)</span>, a publicly available data
collection of annotaed, consistently processed gene expression datasets based
on RNA-sequencing. We downloaded gene expression and metadata as <code>RData</code> objects
from the <a href="https://amp.pharm.mssm.edu/archs4/download.html">ARCHS4 website</a> (version 8.0).<a href="#fn11" class="footnote-ref" id="fnref11"><sup>11</sup></a></p>
<p>We filtered samples by the following criteria (figure <a href="curating-data.html#fig:funnelarchs">3.2</a>):</p>
<ol style="list-style-type: decimal">
<li>The library is a transcriptomic cDNA library, the library strategy is RNA-seq, and either polyA or total RNA were extracted.</li>
<li>No single-cell RNA-seq samples (none of the annotation fields may contain the keywords “single-cell,” “single cell” or “smartseq”)</li>
<li>At least 500,000 reads could be mapped to genes.</li>
<li>The tissue could be mapped to the controlled vocabulary (section <a href="curating-data.html#normalize-tissues">3.1</a>).</li>
<li>Finally, we only retained samples originating from tissues for which a <em>reference signature</em> is available.</li>
</ol>
<p>Gene counts were normalized into TPM values before analysing them with BioQC.</p>
<div class="figure" style="text-align: center"><span id="fig:funnelarchs"></span>
<img src="/home/sturm/projects/2019/bioqc_geo/figures/funnel_archs4.png" alt="Summary of filtering steps on ARCHS4 samples" width="40%" />
<p class="caption">
Figure 3.2: Summary of filtering steps on ARCHS4 samples
</p>
</div>

</div>
</div>
<h3> References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-geoquery" class="csl-entry">
Davis, S., and P. S. Meltzer. 2007. <span>“<span>GEOquery</span>: A Bridge Between the Gene Expression Omnibus (<span>GEO</span>) and <span>BioConductor</span>.”</span> <em>Bioinformatics</em> 23 (14): 1846–47. <a href="https://doi.org/10.1093/bioinformatics/btm254">https://doi.org/10.1093/bioinformatics/btm254</a>.
</div>
<div id="ref-Huber2015" class="csl-entry">
Huber, Wolfgang, Vincent J Carey, Robert Gentleman, Simon Anders, Marc Carlson, Benilton S Carvalho, Hector Corrada Bravo, et al. 2015. <span>“Orchestrating High-Throughput Genomic Analysis with Bioconductor.”</span> <em>Nature Methods</em> 12 (2): 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.
</div>
<div id="ref-Lachmann2018" class="csl-entry">
Lachmann, Alexander, Denis Torre, Alexandra B. Keenan, Kathleen M. Jagodnik, Hoyjin J. Lee, Lily Wang, Moshe C. Silverstein, and Avi Ma’ayan. 2018. <span>“Massive Mining of Publicly Available <span>RNA</span>-Seq Data from Human and Mouse.”</span> <em>Nature Communications</em> 9 (1). <a href="https://doi.org/10.1038/s41467-018-03751-6">https://doi.org/10.1038/s41467-018-03751-6</a>.
</div>
<div id="ref-geometadb" class="csl-entry">
Zhu, Yuelin, Sean Davis, Robert Stephens, Paul S. Meltzer, and Yidong Chen. 2008. <span>“<span class="nocase">GEOmetadb: Powerful alternative search engine for the Gene Expression Omnibus</span>.”</span> <em>Bioinformatics</em> 24 (23): 2798–2800. <a href="https://doi.org/10.1093/bioinformatics/btn520">https://doi.org/10.1093/bioinformatics/btn520</a>.
</div>
</div>
<div class="footnotes">
<hr />
<ol start="9">
<li id="fn9"><p><a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx" class="uri">https://github.com/grst/BioQC_GEO_analysis/blob/master/manual_annotation/tissue_annotation.xlsx</a><a href="curating-data.html#fnref9" class="footnote-back">↩︎</a></p></li>
<li id="fn10"><p><a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R" class="uri">https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R</a><a href="curating-data.html#fnref10" class="footnote-back">↩︎</a></p></li>
<li id="fn11"><p><a href="https://amp.pharm.mssm.edu/archs4/download.html" class="uri">https://amp.pharm.mssm.edu/archs4/download.html</a><a href="curating-data.html#fnref11" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="validating-signatures.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="testing-for-tissue-heterogeneity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
