<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A prevalence of tissue heterogeneity in gene expression data: Supplementary Information</title>
  <meta name="description" content="Systematically testing the GEO for tissue heterogeneity">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="A prevalence of tissue heterogeneity in gene expression data: Supplementary Information" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematically testing the GEO for tissue heterogeneity" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A prevalence of tissue heterogeneity in gene expression data: Supplementary Information" />
  
  <meta name="twitter:description" content="Systematically testing the GEO for tissue heterogeneity" />
  

<meta name="author" content="Gregor Sturm, Jitao David Zhang">


<meta name="date" content="2018-07-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="redo-analysis-for-archs4.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BioQC GEO Analysis</a></li>
<li class="chapter" data-level="2" data-path="validating-signatures.html"><a href="validating-signatures.html"><i class="fa fa-check"></i><b>2</b> Validating Tissue Signatures</a><ul>
<li class="chapter" data-level="2.1" data-path="validating-signatures.html"><a href="validating-signatures.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-validation-of-signatures-on-the-gtex-dataset"><i class="fa fa-check"></i><b>2.2</b> Cross-Validation of signatures on the GTEx dataset</a></li>
<li class="chapter" data-level="2.3" data-path="validating-signatures.html"><a href="validating-signatures.html#solid-signatures"><i class="fa fa-check"></i><b>2.3</b> ‘solid’ signatures</a></li>
<li class="chapter" data-level="2.4" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-platform-cross-species-validation"><i class="fa fa-check"></i><b>2.4</b> Cross-Platform Cross-Species validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="curating-data.html"><a href="curating-data.html"><i class="fa fa-check"></i><b>3</b> Curating Data</a><ul>
<li class="chapter" data-level="3.1" data-path="curating-data.html"><a href="curating-data.html#selecting-samples-by-metadata"><i class="fa fa-check"></i><b>3.1</b> Selecting Samples by Metadata</a></li>
<li class="chapter" data-level="3.2" data-path="curating-data.html"><a href="curating-data.html#normalize-tissues"><i class="fa fa-check"></i><b>3.2</b> Normalize Tissues</a></li>
<li class="chapter" data-level="3.3" data-path="curating-data.html"><a href="curating-data.html#tissue-signatures"><i class="fa fa-check"></i><b>3.3</b> Map tissues to signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="setup-database.html"><a href="setup-database.html"><i class="fa fa-check"></i><b>4</b> Setup Database</a><ul>
<li class="chapter" data-level="4.1" data-path="setup-database.html"><a href="setup-database.html#tables-explained"><i class="fa fa-check"></i><b>4.1</b> Tables explained</a><ul>
<li class="chapter" data-level="4.1.1" data-path="setup-database.html"><a href="setup-database.html#geometadb"><i class="fa fa-check"></i><b>4.1.1</b> GEOmetadb</a></li>
<li class="chapter" data-level="4.1.2" data-path="setup-database.html"><a href="setup-database.html#bioqc"><i class="fa fa-check"></i><b>4.1.2</b> BioQC</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="setup-database.html"><a href="setup-database.html#import-geometadb"><i class="fa fa-check"></i><b>4.2</b> Import GEOmetadb</a><ul>
<li class="chapter" data-level="4.2.1" data-path="setup-database.html"><a href="setup-database.html#fix-foreign-key-constraints"><i class="fa fa-check"></i><b>4.2.1</b> Fix foreign key constraints</a></li>
<li class="chapter" data-level="4.2.2" data-path="setup-database.html"><a href="setup-database.html#extract-tissue-annotation"><i class="fa fa-check"></i><b>4.2.2</b> Extract Tissue annotation</a></li>
<li class="chapter" data-level="4.2.3" data-path="setup-database.html"><a href="setup-database.html#load-annotation-information"><i class="fa fa-check"></i><b>4.2.3</b> Load annotation information</a></li>
<li class="chapter" data-level="4.2.4" data-path="setup-database.html"><a href="setup-database.html#import-summary-statistics-for-each-study"><i class="fa fa-check"></i><b>4.2.4</b> Import summary statistics for each study</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-data"><i class="fa fa-check"></i><b>4.3</b> Import BioQC data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="setup-database.html"><a href="setup-database.html#signatures"><i class="fa fa-check"></i><b>4.3.1</b> Signatures</a></li>
<li class="chapter" data-level="4.3.2" data-path="setup-database.html"><a href="setup-database.html#tissue-annotation"><i class="fa fa-check"></i><b>4.3.2</b> Tissue Annotation</a></li>
<li class="chapter" data-level="4.3.3" data-path="setup-database.html"><a href="setup-database.html#tissue-sets"><i class="fa fa-check"></i><b>4.3.3</b> Tissue Sets</a></li>
<li class="chapter" data-level="4.3.4" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-results"><i class="fa fa-check"></i><b>4.3.4</b> BioQC results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sample-selection.html"><a href="sample-selection.html"><i class="fa fa-check"></i><b>5</b> Sample Selection and processing</a><ul>
<li class="chapter" data-level="5.1" data-path="sample-selection.html"><a href="sample-selection.html#sample-preselection"><i class="fa fa-check"></i><b>5.1</b> Sample Preselection</a><ul>
<li class="chapter" data-level="5.1.1" data-path="sample-selection.html"><a href="sample-selection.html#required-annotation"><i class="fa fa-check"></i><b>5.1.1</b> Required annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="sample-selection.html"><a href="sample-selection.html#sample-processing"><i class="fa fa-check"></i><b>5.2</b> Sample Processing with BioQC</a></li>
<li class="chapter" data-level="5.3" data-path="sample-selection.html"><a href="sample-selection.html#sample-post-selection"><i class="fa fa-check"></i><b>5.3</b> Sample Post-selection</a><ul>
<li class="chapter" data-level="5.3.1" data-path="sample-selection.html"><a href="sample-selection.html#excluding-multi-channel-microarrays"><i class="fa fa-check"></i><b>5.3.1</b> Excluding multi-channel microarrays</a></li>
<li class="chapter" data-level="5.3.2" data-path="sample-selection.html"><a href="sample-selection.html#exclude-non-mapped-tissues"><i class="fa fa-check"></i><b>5.3.2</b> Exclude non-mapped tissues</a></li>
<li class="chapter" data-level="5.3.3" data-path="sample-selection.html"><a href="sample-selection.html#select-organisms"><i class="fa fa-check"></i><b>5.3.3</b> Select organisms</a></li>
<li class="chapter" data-level="5.3.4" data-path="sample-selection.html"><a href="sample-selection.html#quality-control"><i class="fa fa-check"></i><b>5.3.4</b> Quality control</a></li>
<li class="chapter" data-level="5.3.5" data-path="sample-selection.html"><a href="sample-selection.html#tissue-abundance"><i class="fa fa-check"></i><b>5.3.5</b> Tissue abundance</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="sample-selection.html"><a href="sample-selection.html#archs4-rna-seq-data"><i class="fa fa-check"></i><b>5.4</b> ARCHS4 RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html"><i class="fa fa-check"></i><b>6</b> Testing for tissue heterogeneity</a><ul>
<li class="chapter" data-level="6.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:signatures"><i class="fa fa-check"></i><b>6.1</b> Tissue signatures</a></li>
<li class="chapter" data-level="6.2" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:data_selection"><i class="fa fa-check"></i><b>6.2</b> Data</a></li>
<li class="chapter" data-level="6.3" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#testing-samples-for-heterogeneity"><i class="fa fa-check"></i><b>6.3</b> Testing samples for heterogeneity</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>7</b> Results</a></li>
<li class="chapter" data-level="8" data-path="redo-analysis-for-archs4.html"><a href="redo-analysis-for-archs4.html"><i class="fa fa-check"></i><b>8</b> Redo analysis for ARCHS4</a><ul>
<li class="chapter" data-level="8.1" data-path="redo-analysis-for-archs4.html"><a href="redo-analysis-for-archs4.html#quick-plots-to-check-if-everything-is-ok"><i class="fa fa-check"></i><b>8.1</b> Quick plots to check if everything is ok</a></li>
<li class="chapter" data-level="8.2" data-path="redo-analysis-for-archs4.html"><a href="redo-analysis-for-archs4.html#sample-count-by-tissue"><i class="fa fa-check"></i><b>8.2</b> Sample count by tissue</a></li>
<li class="chapter" data-level="8.3" data-path="redo-analysis-for-archs4.html"><a href="redo-analysis-for-archs4.html#same-analysis"><i class="fa fa-check"></i><b>8.3</b> same analysis…</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="figures-for-publication.html"><a href="figures-for-publication.html"><i class="fa fa-check"></i><b>9</b> Figures for publication</a><ul>
<li class="chapter" data-level="9.1" data-path="figures-for-publication.html"><a href="figures-for-publication.html#contamination-heatmaps"><i class="fa fa-check"></i><b>9.1</b> Contamination heatmaps</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A prevalence of tissue heterogeneity in gene expression data: Supplementary Information</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figures-for-publication" class="section level1">
<h1><span class="header-section-number">9</span> Figures for publication</h1>
<div id="contamination-heatmaps" class="section level2">
<h2><span class="header-section-number">9.1</span> Contamination heatmaps</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&#39;../results/data_processed.RData&#39;</span>)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&#39;../results/models.RData&#39;</span>)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"></a>
<a class="sourceLine" id="cb34-4" data-line-number="4">contam_count_array =<span class="st"> </span>data_corr <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-5" data-line-number="5"><span class="st">  </span><span class="kw">filter</span>(pcorr_adj <span class="op">&lt;</span><span class="st"> </span>P_THRES) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-6" data-line-number="6"><span class="st">  </span><span class="kw">group_by</span>(TGROUP, SIGNATURE) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-7" data-line-number="7"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cnt =</span> <span class="kw">n_distinct</span>(GSM))</a>
<a class="sourceLine" id="cb34-8" data-line-number="8">data_corr_array =<span class="st"> </span>data_corr</a>
<a class="sourceLine" id="cb34-9" data-line-number="9"></a>
<a class="sourceLine" id="cb34-10" data-line-number="10"><span class="kw">load</span>(<span class="st">&#39;../results/archs4/archs4_data_processed.RData&#39;</span>)</a>
<a class="sourceLine" id="cb34-11" data-line-number="11"><span class="kw">load</span>(<span class="st">&#39;../results/archs4/archs4_models.RData&#39;</span>)</a>
<a class="sourceLine" id="cb34-12" data-line-number="12"></a>
<a class="sourceLine" id="cb34-13" data-line-number="13">contam_count_ngs =<span class="st"> </span>data_corr <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-14" data-line-number="14"><span class="st">  </span><span class="kw">filter</span>(pcorr_adj <span class="op">&lt;</span><span class="st"> </span>P_THRES) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-15" data-line-number="15"><span class="st">  </span><span class="kw">group_by</span>(TGROUP, SIGNATURE) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-16" data-line-number="16"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cnt =</span> <span class="kw">n_distinct</span>(GSM))</a>
<a class="sourceLine" id="cb34-17" data-line-number="17">data_corr_ngs =<span class="st"> </span>data_corr</a></code></pre></div>
<p>same as in the section before, but all bioqc signatures aggregated by tissue groups.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">bioqc_tissue_set =<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;../data/bioqc_geo_oracle_dump/BIOQC_TISSUE_SET_DATA_TABLE.csv&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(TISSUE_SET <span class="op">==</span><span class="st"> &#39;bioqc_all&#39;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="st">  </span><span class="kw">select</span>(<span class="dt">SIG_ID=</span>SIGNATURE, <span class="dt">DETECTED_TGROUP=</span>TGROUP)</a></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   SIGNATURE = col_integer(),
##   TISSUE = col_character(),
##   TGROUP = col_character(),
##   TISSUE_SET = col_character()
## )</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1">bioqc_all =<span class="st"> </span>selected_signatures <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(<span class="kw">str_detect</span>(SIGNATURE, <span class="st">&#39;BIOQC&#39;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-3" data-line-number="3"><span class="st">  </span><span class="kw">pull</span>(<span class="st">&quot;SIGNATURE&quot;</span>)</a>
<a class="sourceLine" id="cb37-4" data-line-number="4"></a>
<a class="sourceLine" id="cb37-5" data-line-number="5">contam_count_ngs_tgroup =<span class="st"> </span>data_corr_ngs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-6" data-line-number="6"><span class="st">  </span><span class="kw">inner_join</span>(bioqc_signatures, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;SIGNATURE&quot;</span>=<span class="st">&quot;SIG_NAME&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-7" data-line-number="7"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">SIG_ID=</span>ID) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-8" data-line-number="8"><span class="st">  </span><span class="kw">inner_join</span>(bioqc_tissue_set) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-9" data-line-number="9"><span class="st">  </span><span class="kw">filter</span>(pcorr_adj <span class="op">&lt;</span><span class="st"> </span>P_THRES) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-10" data-line-number="10"><span class="st">  </span><span class="kw">group_by</span>(TGROUP, DETECTED_TGROUP) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-11" data-line-number="11"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cnt =</span> <span class="kw">n_distinct</span>(GSM))</a></code></pre></div>
<pre><code>## Joining, by = &quot;SIG_ID&quot;</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">contam_count_array_tgroup =<span class="st"> </span>data_corr_array <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="st">  </span><span class="kw">inner_join</span>(bioqc_signatures, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;SIGNATURE&quot;</span>=<span class="st">&quot;SIG_NAME&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-3" data-line-number="3"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">SIG_ID=</span>ID) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-4" data-line-number="4"><span class="st">  </span><span class="kw">inner_join</span>(bioqc_tissue_set) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-5" data-line-number="5"><span class="st">  </span><span class="kw">filter</span>(pcorr_adj <span class="op">&lt;</span><span class="st"> </span>P_THRES) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-6" data-line-number="6"><span class="st">  </span><span class="kw">group_by</span>(TGROUP, DETECTED_TGROUP) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-7" data-line-number="7"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cnt =</span> <span class="kw">n_distinct</span>(GSM))</a></code></pre></div>
<pre><code>## Joining, by = &quot;SIG_ID&quot;</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">tgroups =<span class="st"> </span>bioqc_tissue_set<span class="op">$</span>DETECTED_TGROUP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>()</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">contam_count_ngs_tgroup <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb41-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">factor</span>(DETECTED_TGROUP, <span class="dt">levels=</span>tgroups), <span class="dt">y=</span>TGROUP)) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-4" data-line-number="4"><span class="st">      </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> cnt <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-5" data-line-number="5"><span class="st">      </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>cnt)) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-6" data-line-number="6"><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-7" data-line-number="7"><span class="st">      </span><span class="kw">scale_x_discrete</span>(<span class="dt">drop=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-8" data-line-number="8"><span class="st">      </span><span class="kw">ylab</span>(<span class="st">&quot;reference tissue&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-9" data-line-number="9"><span class="st">      </span><span class="kw">xlab</span>(<span class="st">&quot;detected signature&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-10" data-line-number="10"><span class="st">      </span><span class="kw">background_grid</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb41-11" data-line-number="11"><span class="st">      </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#ececff&quot;</span>))</a></code></pre></div>
<p><img src="95_figures_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">tgroups =<span class="st"> </span>bioqc_tissue_set<span class="op">$</span>DETECTED_TGROUP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unique</span>()</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">contam_count_array_tgroup <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">factor</span>(DETECTED_TGROUP, <span class="dt">levels=</span>tgroups), <span class="dt">y=</span>TGROUP)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-4" data-line-number="4"><span class="st">      </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> cnt <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="st">      </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>cnt)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6"><span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-7" data-line-number="7"><span class="st">      </span><span class="kw">scale_x_discrete</span>(<span class="dt">drop=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-8" data-line-number="8"><span class="st">      </span><span class="kw">ylab</span>(<span class="st">&quot;reference tissue&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-9" data-line-number="9"><span class="st">      </span><span class="kw">xlab</span>(<span class="st">&quot;detected signature&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-10" data-line-number="10"><span class="st">      </span><span class="kw">background_grid</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb42-11" data-line-number="11"><span class="st">      </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#ececff&quot;</span>))</a></code></pre></div>
<p><img src="95_figures_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>

<div id="refs" class="references">
<div>
<p>Gönen, Mithat. 2009. “Statistical aspects of gene signatures and molecular targets.” <em>Gastrointest Cancer Res</em> 3 (2 Suppl): 19–21. <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2684735{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract" class="uri">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2684735{\&amp;}tool=pmcentrez{\&amp;}rendertype=abstract</a>.</p>
</div>
</div>
</div>
</div>






            </section>

          </div>
        </div>
      </div>
<a href="redo-analysis-for-archs4.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
