<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A prevalence of tissue heterogeneity in gene expression data: Supplementary Information</title>
  <meta name="description" content="Systematically testing the GEO for tissue heterogeneity">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="A prevalence of tissue heterogeneity in gene expression data: Supplementary Information" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematically testing the GEO for tissue heterogeneity" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A prevalence of tissue heterogeneity in gene expression data: Supplementary Information" />
  
  <meta name="twitter:description" content="Systematically testing the GEO for tissue heterogeneity" />
  

<meta name="author" content="Gregor Sturm, Jitao David Zhang">


<meta name="date" content="2018-09-22">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="sample-selection.html">
<link rel="next" href="results.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BioQC GEO Analysis</a></li>
<li class="chapter" data-level="2" data-path="validating-signatures.html"><a href="validating-signatures.html"><i class="fa fa-check"></i><b>2</b> Validating Tissue Signatures</a><ul>
<li class="chapter" data-level="2.1" data-path="validating-signatures.html"><a href="validating-signatures.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-validation-of-signatures-on-the-gtex-dataset"><i class="fa fa-check"></i><b>2.2</b> Cross-Validation of signatures on the GTEx dataset</a></li>
<li class="chapter" data-level="2.3" data-path="validating-signatures.html"><a href="validating-signatures.html#solid-signatures"><i class="fa fa-check"></i><b>2.3</b> ‘solid’ signatures</a></li>
<li class="chapter" data-level="2.4" data-path="validating-signatures.html"><a href="validating-signatures.html#cross-platform-cross-species-validation"><i class="fa fa-check"></i><b>2.4</b> Cross-Platform Cross-Species validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="curating-data.html"><a href="curating-data.html"><i class="fa fa-check"></i><b>3</b> Curating Data</a><ul>
<li class="chapter" data-level="3.1" data-path="curating-data.html"><a href="curating-data.html#selecting-samples-by-metadata"><i class="fa fa-check"></i><b>3.1</b> Selecting Samples by Metadata</a></li>
<li class="chapter" data-level="3.2" data-path="curating-data.html"><a href="curating-data.html#normalize-tissues"><i class="fa fa-check"></i><b>3.2</b> Normalize Tissues</a></li>
<li class="chapter" data-level="3.3" data-path="curating-data.html"><a href="curating-data.html#tissue-signatures"><i class="fa fa-check"></i><b>3.3</b> Map tissues to signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="setup-database.html"><a href="setup-database.html"><i class="fa fa-check"></i><b>4</b> Setup Database</a><ul>
<li class="chapter" data-level="4.1" data-path="setup-database.html"><a href="setup-database.html#tables-explained"><i class="fa fa-check"></i><b>4.1</b> Tables explained</a><ul>
<li class="chapter" data-level="4.1.1" data-path="setup-database.html"><a href="setup-database.html#geometadb"><i class="fa fa-check"></i><b>4.1.1</b> GEOmetadb</a></li>
<li class="chapter" data-level="4.1.2" data-path="setup-database.html"><a href="setup-database.html#bioqc"><i class="fa fa-check"></i><b>4.1.2</b> BioQC</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="setup-database.html"><a href="setup-database.html#import-geometadb"><i class="fa fa-check"></i><b>4.2</b> Import GEOmetadb</a><ul>
<li class="chapter" data-level="4.2.1" data-path="setup-database.html"><a href="setup-database.html#fix-foreign-key-constraints"><i class="fa fa-check"></i><b>4.2.1</b> Fix foreign key constraints</a></li>
<li class="chapter" data-level="4.2.2" data-path="setup-database.html"><a href="setup-database.html#extract-tissue-annotation"><i class="fa fa-check"></i><b>4.2.2</b> Extract Tissue annotation</a></li>
<li class="chapter" data-level="4.2.3" data-path="setup-database.html"><a href="setup-database.html#load-annotation-information"><i class="fa fa-check"></i><b>4.2.3</b> Load annotation information</a></li>
<li class="chapter" data-level="4.2.4" data-path="setup-database.html"><a href="setup-database.html#import-summary-statistics-for-each-study"><i class="fa fa-check"></i><b>4.2.4</b> Import summary statistics for each study</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-data"><i class="fa fa-check"></i><b>4.3</b> Import BioQC data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="setup-database.html"><a href="setup-database.html#signatures"><i class="fa fa-check"></i><b>4.3.1</b> Signatures</a></li>
<li class="chapter" data-level="4.3.2" data-path="setup-database.html"><a href="setup-database.html#tissue-annotation"><i class="fa fa-check"></i><b>4.3.2</b> Tissue Annotation</a></li>
<li class="chapter" data-level="4.3.3" data-path="setup-database.html"><a href="setup-database.html#tissue-sets"><i class="fa fa-check"></i><b>4.3.3</b> Tissue Sets</a></li>
<li class="chapter" data-level="4.3.4" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-results"><i class="fa fa-check"></i><b>4.3.4</b> BioQC results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sample-selection.html"><a href="sample-selection.html"><i class="fa fa-check"></i><b>5</b> Sample Selection and processing</a><ul>
<li class="chapter" data-level="5.1" data-path="sample-selection.html"><a href="sample-selection.html#sample-preselection"><i class="fa fa-check"></i><b>5.1</b> Sample Preselection</a><ul>
<li class="chapter" data-level="5.1.1" data-path="sample-selection.html"><a href="sample-selection.html#required-annotation"><i class="fa fa-check"></i><b>5.1.1</b> Required annotation</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="sample-selection.html"><a href="sample-selection.html#sample-processing"><i class="fa fa-check"></i><b>5.2</b> Sample Processing with BioQC</a></li>
<li class="chapter" data-level="5.3" data-path="sample-selection.html"><a href="sample-selection.html#sample-post-selection"><i class="fa fa-check"></i><b>5.3</b> Sample Post-selection</a><ul>
<li class="chapter" data-level="5.3.1" data-path="sample-selection.html"><a href="sample-selection.html#excluding-multi-channel-microarrays"><i class="fa fa-check"></i><b>5.3.1</b> Excluding multi-channel microarrays</a></li>
<li class="chapter" data-level="5.3.2" data-path="sample-selection.html"><a href="sample-selection.html#exclude-non-mapped-tissues"><i class="fa fa-check"></i><b>5.3.2</b> Exclude non-mapped tissues</a></li>
<li class="chapter" data-level="5.3.3" data-path="sample-selection.html"><a href="sample-selection.html#select-organisms"><i class="fa fa-check"></i><b>5.3.3</b> Select organisms</a></li>
<li class="chapter" data-level="5.3.4" data-path="sample-selection.html"><a href="sample-selection.html#quality-control"><i class="fa fa-check"></i><b>5.3.4</b> Quality control</a></li>
<li class="chapter" data-level="5.3.5" data-path="sample-selection.html"><a href="sample-selection.html#tissue-abundance"><i class="fa fa-check"></i><b>5.3.5</b> Tissue abundance</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="sample-selection.html"><a href="sample-selection.html#archs4-rna-seq-data"><i class="fa fa-check"></i><b>5.4</b> ARCHS4 RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html"><i class="fa fa-check"></i><b>6</b> Testing for tissue heterogeneity</a><ul>
<li class="chapter" data-level="6.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:signatures"><i class="fa fa-check"></i><b>6.1</b> Tissue signatures</a></li>
<li class="chapter" data-level="6.2" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#sec:data_selection"><i class="fa fa-check"></i><b>6.2</b> Data</a></li>
<li class="chapter" data-level="6.3" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#testing-samples-for-heterogeneity"><i class="fa fa-check"></i><b>6.3</b> Testing samples for heterogeneity</a><ul>
<li class="chapter" data-level="6.3.1" data-path="testing-for-tissue-heterogeneity.html"><a href="testing-for-tissue-heterogeneity.html#determining-tau."><i class="fa fa-check"></i><b>6.3.1</b> Determining <span class="math inline">\(\tau\)</span>.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>7</b> Results</a><ul>
<li class="chapter" data-level="7.1" data-path="results.html"><a href="results.html#plots-of-scores-for-manual-inspection"><i class="fa fa-check"></i><b>7.1</b> Plots of scores for manual inspection</a></li>
<li class="chapter" data-level="7.2" data-path="results.html"><a href="results.html#heterogeneity-using-gtex-signatures-only"><i class="fa fa-check"></i><b>7.2</b> Heterogeneity using GTEX signatures only</a></li>
<li class="chapter" data-level="7.3" data-path="results.html"><a href="results.html#heterogeneity-using-all-signatures"><i class="fa fa-check"></i><b>7.3</b> Heterogeneity using all signatures</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="figures-for-publication.html"><a href="figures-for-publication.html"><i class="fa fa-check"></i><b>8</b> Figures for publication</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A prevalence of tissue heterogeneity in gene expression data: Supplementary Information</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="testing-for-tissue-heterogeneity" class="section level1">
<h1><span class="header-section-number">6</span> Testing for tissue heterogeneity</h1>
<div id="sec:signatures" class="section level2">
<h2><span class="header-section-number">6.1</span> Tissue signatures</h2>
<p>In section <a href="validating-signatures.html#validating-signatures">2</a>, we identified a set of
9 <em>reference signatures</em> (table <a href="testing-for-tissue-heterogeneity.html#tab:refsig">6.1</a>) which unambiguously
identify their corresponding tissue across platforms and species.</p>
<table>
<caption><span id="tab:refsig">Table 6.1: </span>reference signatures</caption>
<thead>
<tr class="header">
<th align="left">Reference Signature</th>
<th align="left">Tissue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GTEX_Blood</td>
<td align="left">blood</td>
</tr>
<tr class="even">
<td align="left">GTEX_Brain</td>
<td align="left">brain</td>
</tr>
<tr class="odd">
<td align="left">GTEX_Heart</td>
<td align="left">heart</td>
</tr>
<tr class="even">
<td align="left">GTEX_Kidney</td>
<td align="left">kidney</td>
</tr>
<tr class="odd">
<td align="left">GTEX_Liver</td>
<td align="left">liver</td>
</tr>
<tr class="even">
<td align="left">GTEX_Muscle_Skeletal</td>
<td align="left">skeletal muscle</td>
</tr>
<tr class="odd">
<td align="left">GTEX_Pancreas</td>
<td align="left">pancreas</td>
</tr>
<tr class="even">
<td align="left">GTEX_Skin</td>
<td align="left">skin</td>
</tr>
<tr class="odd">
<td align="left">GTEX_Testis</td>
<td align="left">testis</td>
</tr>
</tbody>
</table>
</div>
<div id="sec:data_selection" class="section level2">
<h2><span class="header-section-number">6.2</span> Data</h2>
<p>In section <a href="sample-selection.html#sample-selection">5</a> we show in detail, how we obtain
and filter data from the Gene Expression Omnibus.
Table <a href="testing-for-tissue-heterogeneity.html#tab:filteringprocess">6.2</a> shows a summary of this process, listing
the remaining samples and studies after each filtering step.</p>
<table>
<caption><span id="tab:filteringprocess">Table 6.2: </span>Selecting samples from GEO</caption>
<thead>
<tr class="header">
<th align="left">filtering step</th>
<th align="right">samples</th>
<th align="right">studies</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">total (as of 2016-12-07)</td>
<td align="right">1945417</td>
<td align="right">73719</td>
</tr>
<tr class="even">
<td align="left">tissue annotated</td>
<td align="right">760798</td>
<td align="right">24267</td>
</tr>
<tr class="odd">
<td align="left">gene symbols annotated</td>
<td align="right">275206</td>
<td align="right">9632</td>
</tr>
<tr class="even">
<td align="left">human orthologous gene symbols</td>
<td align="right">253210</td>
<td align="right">8083</td>
</tr>
<tr class="odd">
<td align="left">Single-channel microarray</td>
<td align="right">235237</td>
<td align="right">7561</td>
</tr>
<tr class="even">
<td align="left">tissue mapped to controlled vocabulary</td>
<td align="right">136230</td>
<td align="right">3800</td>
</tr>
<tr class="odd">
<td align="left">human, rat, mouse</td>
<td align="right">134334</td>
<td align="right">3757</td>
</tr>
<tr class="even">
<td align="left">minimal inter-sample variance</td>
<td align="right">96074</td>
<td align="right">3129</td>
</tr>
<tr class="odd">
<td align="left">housekeeping signatures</td>
<td align="right">81845</td>
<td align="right">2926</td>
</tr>
<tr class="even">
<td align="left">reference signature available</td>
<td align="right">51948</td>
<td align="right">1760</td>
</tr>
</tbody>
</table>
</div>
<div id="testing-samples-for-heterogeneity" class="section level2">
<h2><span class="header-section-number">6.3</span> Testing samples for heterogeneity</h2>
<p>We tested for enrichment of
120
selected signatures provided by BioQC, the
9
reference signatures generated by us and one random control signature of
100 randomly drawn genes on all 51948 selected samples
resulting in a list of 6805188 (sample, signature, pvalue) pairs.
This process is described in detail in section <a href="sample-selection.html#sample-processing">5.2</a>.</p>
<p>Our intention is to identify samples that show <em>tissue heterogeneity</em>,
<em>i.e.</em> unintentional profiling of cells of other origin than the target tissue
of profiling. We classify samples as <em>not heterogenous</em> or <em>heterogenous</em>.
We call a classification <em>true-positive</em> if the given sample is classified
as <em>heterogenous</em> and the sample indeed contains cells different from the
annotated tissues. Analogous, we call a classification <em>false-positive</em>
if the given sample is classified as <em>heterogenous</em> but in reality only contains cells from the annotated tissue.</p>
<p>Naively, we would label a sample as heterogenous, if a signature unrelated
to the annotated tissue exceeds a certain score. The problem with this
approach is, that some signatures overlap; the resulting scores are therefore
correlated and will lead to false-positives. One cannot simply solve this
problem by excuding genes that are members of multiple signatures, as it is
easily possible to build two (in fact many) distinct, non-overlapping
signatures matching the same tissue.</p>
<p>In section <a href="validating-signatures.html#validating-signatures">2</a> we haven created toroughly validated
<em>reference signatures</em> for 9 tissues. Even though we have demonstrated that
each signature unambigously identifies its corresponding tissue (<em>i.e.</em> scores
highest), the signatures could still be correlated. Some of them in fact are,
e.g. cardiac muscle and skeletal muscle (see figure
<a href="testing-for-tissue-heterogeneity.html#fig:correlationexample">6.1</a>). Moreover, we lack sufficient data to perform an
independent-sample validation on the signatures provided by BioQC. Accounting
for correlation with the reference signatures, we can still avoid
false-positives, even if a signature is characteristic for a tissue
histologically close to the annotated tissue.</p>
<p>We therefore take the following approach to account for the correlation of signatures:
A given sample <span class="math inline">\(s\)</span> annotated as tissue <span class="math inline">\(t\)</span> is tested for enrichment with signature <span class="math inline">\(k_{\text{test}}\)</span> resulting in a p-value <span class="math inline">\(p_{\text{test}}\)</span>. Let <span class="math inline">\(k_{\text{ref}}\)</span> be the reference signature associated with tissue <span class="math inline">\(t\)</span> and <span class="math inline">\(p_{\text{ref}}\)</span> the p-value of testing <span class="math inline">\(s\)</span> for enrichment of <span class="math inline">\(k_{\text{ref}}\)</span>. Let <span class="math inline">\(\tau\)</span> be a certain false discovery rate (FDR)-threshold.</p>
<ol style="list-style-type: decimal">
<li>If the Benjamini-Hochberg (BH)-adjusted <span class="math inline">\(p_{\text{test}} &gt;= \tau\)</span>, we
assume that <span class="math inline">\(s\)</span> is not heterogenous; else continue.</li>
<li>We fit a linear model using <code>rlm</code> from the <code>R</code> <code>MASS</code> package of
<span class="math inline">\(|log10(p_{\text{test}})|\)</span> against <span class="math inline">\(|log10(p_{\text{ref}})|\)</span> for all samples
annotated as <span class="math inline">\(t\)</span>. We assume that the residuals <span class="math inline">\(R\)</span> of the linear model follow
a normal distribution <span class="math inline">\(R \sim \mathcal{N}(0, \sigma^2)\)</span>, where <span class="math inline">\(\sigma\)</span> is the
standard deviation of the residuals.</li>
<li>We extract the residual <span class="math inline">\(r\)</span> corresponding to sample <span class="math inline">\(s\)</span>. We calculate the
p-value <span class="math inline">\(p_{\text{corr}} = 1 - \text{CDF}_R(r)\)</span> where <span class="math inline">\(\text{CDF}_R\)</span> is the
cumulative density function of <span class="math inline">\(\mathcal{N}(0, \sigma^2)\)</span>.</li>
<li>If the BH-adjusted <span class="math inline">\(p_{\text{corr}} &lt; \tau\)</span>, we reject the
hypothesis that <span class="math inline">\(k_{\text{test}}\)</span> is enriched only due to correlation and label
the sample as heterogenous.</li>
</ol>
<div id="determining-tau." class="section level3">
<h3><span class="header-section-number">6.3.1</span> Determining <span class="math inline">\(\tau\)</span>.</h3>
<p>We desire an overall FDR of 0.01. Since we test in a two-step procedure, we choose a threshold <span class="math inline">\(\tau\)</span>, such that the
overall FDR equals 0.01.</p>
<p><span class="math display">\[
\text{FDR} = \tau + (1-\tau) \tau
\]</span>
Explanation:
Of all positives in step 1, there are, per definition, a fraction of <span class="math inline">\(\tau\)</span> false-positives and <span class="math inline">\(1-\tau\)</span> true-positives.
To all positives, the step 2 test is applied. In addition to the false-positives from step 1, there is a fraction of <span class="math inline">\(\tau(1-\tau)\)</span> false-positives among the true-positives from step 1.</p>
<p>Solving the equation yields:
<span class="math display">\[
0.01 = \tau + (1-\tau)\tau \Leftrightarrow 0 = \tau^2 - 2\tau + 0.01
\]</span></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">tau =<span class="st"> </span><span class="kw">polyroot</span>(<span class="kw">c</span>(FDR_THRES, <span class="dv">-2</span>, <span class="dv">1</span>))[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">tau</a></code></pre></div>
<pre><code>## [1] 0.005012563-0i</code></pre>
<div class="figure"><span id="fig:correlationexample"></span>
<img src="40_heterogeneity_test_files/figure-html/correlationexample-1.png" alt="Examples of signature correlation. Panels A-C: correlation of the signature scores (y-axis) against the scores of a reference signature (x-axis). The purple line indicates the model fitted to the data, the shaded area the 3*sigma confidence interval. Points are colored according to whether they would be discarded already in the FDR-filtering step. (A) Brain scores of kidney samples against the scores of the kidney signature. There is virtually no correlation, nor are there outliers. Appraently, no kidney samples are contaminated with neuronal cells. (B) Liver scores of kidney samples against scores of the kidney signature. The samples are not correlated, however some outliers are detected which are samples potentially containing liver cells. (C) Cardiac muscle scores of skeletal muscle samples against skeletal muscle scores. The scores are highly correlated. While most of the poinst exceed the FDR threshold, they will not be classified as outliers because the elevated score can be explained with the correlation. Panels D and E show the boxplots of the scores of various signatures on kidney and heart samples respectively. " width="1152" />
<p class="caption">
Figure 6.1: Examples of signature correlation. Panels A-C: correlation of the signature scores (y-axis) against the scores of a reference signature (x-axis). The purple line indicates the model fitted to the data, the shaded area the 3*sigma confidence interval. Points are colored according to whether they would be discarded already in the FDR-filtering step. (A) Brain scores of kidney samples against the scores of the kidney signature. There is virtually no correlation, nor are there outliers. Appraently, no kidney samples are contaminated with neuronal cells. (B) Liver scores of kidney samples against scores of the kidney signature. The samples are not correlated, however some outliers are detected which are samples potentially containing liver cells. (C) Cardiac muscle scores of skeletal muscle samples against skeletal muscle scores. The scores are highly correlated. While most of the poinst exceed the FDR threshold, they will not be classified as outliers because the elevated score can be explained with the correlation. Panels D and E show the boxplots of the scores of various signatures on kidney and heart samples respectively.
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sample-selection.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
